klippy/extras/pca9533.py

```diff
@@ -6,48 +6,32 @@
 import logging
 from . import bus
 
+BACKGROUND_PRIORITY_CLOCK = 0x7fffffff00000000
+
+PCA9533_PWM0=0b010
+PCA9533_PWM1=0b100
 PCA9533_PLS0=0b101
 
 class PCA9533:
     def __init__(self, config):
         self.printer = config.get_printer()
         self.i2c = bus.MCU_I2C_from_config(config, default_addr=98)
-        #i2c_addr = self.i2c.get_i2c_address()
-
-        name = config.get_name().split()[1]
-        self.gcode = self.printer.lookup_object('gcode')
-        self.gcode.register_mux_command("SET_LED","LED",name,self.set_led,
-            desc="Set the color of an LED")
-
-        self.led0 = config.getint("initial_RED",0,0,1)
-        self.led1 = config.getint("initial_GREEN",0,0,1)
-        self.led2 = config.getint("initial_BLUE",0,0,1)
-        self.led3 = config.getint("initial_WHITE",0,0,1)
-
-        ls0 = (self.led3<<6) | (self.led2<<4) | (self.led1<<2) | (self.led0)
-
-        self.i2c.i2c_write([PCA9533_PLS0,ls0])
-    def set_led(self, gcmd):
-        if gcmd.get_float("RED",self.led0,0,1) != 0:
-            self.led0 = 1
-        else:
-            self.led0 = 0
-        if gcmd.get_float("GREEN",self.led1,0,1) != 0:
-            self.led1 = 1
-        else:
-            self.led1 = 0
-        if gcmd.get_float("BLUE",self.led2,0,1) != 0:
-            self.led2 = 1
-        else:
-            self.led2 = 0
-        if gcmd.get_float("WHITE",self.led3,0,1) != 0:
-            self.led3 = 1
-        else:
-            self.led3 = 0
-        ls0 = (self.led3<<6) | (self.led2<<4) | (self.led1<<2) | (self.led0)
-
-        self.i2c.i2c_write([PCA9533_PLS0,ls0])
-
+        pled = self.printer.load_object(config, "led")
+        self.led_helper = pled.setup_helper(config, self.update_leds, 1, True)
+        self.i2c.i2c_write([PCA9533_PWM0, 85])
+        self.i2c.i2c_write([PCA9533_PWM1, 170])
+        self.update_leds(self.led_helper.get_status()['color_data'], None)
+    def update_leds(self, led_state, print_time):
+        rmap = [0, 2, 3, 1]
+        red, green, blue, white = [rmap[int(v * 3. + .5)] for v in led_state[0]]
```

Comment:
int(v * 3. + .5) create some weird steps (0 to 0.16) (0.17 to 0.5) (0.51 to 0.83) (0.84 to 1)
with
rmap = [0, 2, 3, 1, 1]
and
int(v * 4 + 0)
This will create more intuitive steps (0 to 0.24) (0.25 to 0.49) (0.5 to 0.74) (0.75 to 1)